Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -10,7 +10,22 @@ public Pawn(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var moves = new List<Square>();
+            var currentSquare = board.FindPiece(this);
+            var playerDirection = this.Player == Player.White ? -1 : 1;
+            var potentialMove = Square.At(currentSquare.Row + playerDirection, currentSquare.Col);
+            
+            if (board.GetPiece(potentialMove) == null)
+            {
+                moves.AddIfOnBoard(potentialMove);
+                potentialMove = Square.At(currentSquare.Row + 2 * playerDirection, currentSquare.Col);
```

Comment:
Does this need to be inside the `if`? If so, I'd prefer to invert the `if` statement and return early if it's true instead. So:
```
if (board.GetPiece(potentialMove) != null)
{
  return moves;
}
...
```
to avoid nesting 🙂 

Also I'd recommend creating a new descriptively named variable instead of reassigning to the same variable. I prefer to avoid reassigning in a method as it can make the logic harder to understand. For example in this case you could have a `moveOneStep` variable and a `moveTwoSteps` variable 🙂